Azure/azure-cli
# [RDBMS] `az postgres flexible-server upgrade`: Add major version upgrade for PostgreSQL flexible server

**Related command**
`az postgres flexible-server upgrade`

**Description**
This PR enables major version upgrades for PostgreSQL flexible server.

**Testing Guide**
Example of how to create a server with version 11 and update it to version 14, passing through versions 12 and 13:
- `az postgres flexible-server create -g testgroup -n testsvr --public-access none --tier GeneralPurpose --sku-name Standard_D2ds_v4 --location eastus --version 11`
- `az postgres flexible-server upgrade -g testgroup -n testsvr --version 12 --yes`
- `az postgres flexible-server upgrade -g testgroup -n testsvr --version 13 --yes`
- `az postgres flexible-server upgrade -g testgroup -n testsvr --version 14 --yes`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
